frontend/class-frontend.php

```diff
@@ -2011,42 +2015,106 @@ private function fetch_social_profiles() {
 	 *
 	 * @since 1.5.7
 	 *
-	 * @link  https://developers.google.com/webmasters/structured-data/slsb-overview
+	 * @link  https://developers.google.com/structured-data/site-name
 	 */
-	public function internal_search() {
+	public function website() {
+		$output = '{ "@context": "http://schema.org",
+			"@type": "WebSite",
+			"url": "' . esc_attr( $this->get_home_url() ) . '",
+			"name": "' . esc_attr( $this->get_website_name() ) . '"';
+
+		$output .= $this->add_alternate_name();
+		$output .= $this->internal_search_section();
+
+		$output .= '}';
+
+		$this->json_ld_output( $output );
+	}
+
+	/**
+	 * Retrieves the home URL
+	 *
+	 * @return string
+	 */
+	private function get_home_url() {
+		return trailingslashit( home_url() );
+	}
+
+	/**
+	 * Returns an alternate name if one was specified in the WP SEO settings
+	 *
+	 * @return string
+	 */
+	private function add_alternate_name() {
+		if ( '' !== $this->options['alternate_website_name'] ) {
```

Comment:
YODA you mean? For empty strings I find it easier, we don't enforce it though :)